api/s2n.h

```diff
@@ -1112,6 +1112,18 @@ typedef enum {
  * Sets up a connection to request the certificate status of a peer during an SSL handshake. If set
  * to S2N_STATUS_REQUEST_NONE, no status request is made.
  *
+ * @note SHA-1 is the only supported hash algorithm for the `certID` field. This is different 
+ * from the hash algorithm used for the OCSP signature. If a different hash algorithm is used, 
+ * validation will fail.
+ * If the underlying libcrypto supports OCSP validation, the OCSP stapling information will 
+ * be automatically validated. Users can disable this OCSP verification by calling 
+ * `s2n_config_set_check_stapled_ocsp_response()` with "0" to turn OCSP validation off. Users may 
+ * use `s2n_connection_get_ocsp_response()` together with OCSP validation turned off to retrieve 
+ * the received OCSP stapling information for manual verification. This workaround is however 
+ * unlikely to be needed in most scenarios, as SHA-1 is typically used in the `certID` field.
+ * For more details about the `certID` field, refer to 
+ * [RFC6960](https://datatracker.ietf.org/doc/html/rfc6960#section-4.1.1).
```

Comment:
I think this sounds good. I tried to make it a bit more concise, though, if you want:
```suggestion
 * @note SHA-1 is the only supported hash algorithm for the `certID` field. This is different
 * from the hash algorithm used for the OCSP signature. See 
 * [RFC 6960](https://datatracker.ietf.org/doc/html/rfc6960#section-4.1.1) for more information.
 * While unlikely to be the case, if support for a different hash algorithm is required, the
 * s2n-tls validation can be disabled with `s2n_config_set_check_stapled_ocsp_response()` and the
 * response can be retrieved for manual validation with `s2n_connection_get_ocsp_response()`.
```